client/rest-high-level/src/test/java/org/elasticsearch/client/CrudIT.java

```diff
@@ -141,4 +149,123 @@ public void testGet() throws IOException {
             assertEquals("value1", sourceAsMap.get("field1"));
         }
     }
+
+    public void testIndex() throws IOException {
+        {
+            for (XContentType xContentType : XContentType.values()) {
+                IndexRequest indexRequest = new IndexRequest("index", "type");
+                indexRequest.source(XContentBuilder.builder(xContentType.xContent()).startObject().field("test", "test").endObject());
+
+                IndexResponse indexResponse = execute(indexRequest, highLevelClient()::index, highLevelClient()::indexAsync);
+                assertEquals(RestStatus.CREATED, indexResponse.status());
+                assertEquals(DocWriteResponse.Result.CREATED, indexResponse.getResult());
+                assertEquals("index", indexResponse.getIndex());
+                assertEquals("type", indexResponse.getType());
+                assertTrue(Strings.hasLength(indexResponse.getId()));
+                assertEquals(1L, indexResponse.getVersion());
+                assertNotNull(indexResponse.getShardId());
+                assertEquals(-1, indexResponse.getShardId().getId());
+                assertEquals("index", indexResponse.getShardId().getIndexName());
+                assertEquals("index", indexResponse.getShardId().getIndex().getName());
+                assertEquals("_na_", indexResponse.getShardId().getIndex().getUUID());
+                assertNotNull(indexResponse.getShardInfo());
+                assertEquals(0, indexResponse.getShardInfo().getFailed());
+                assertTrue(indexResponse.getShardInfo().getSuccessful() > 0);
+                assertTrue(indexResponse.getShardInfo().getTotal() > 0);
+            }
+        }
+
+        final XContentType xContentType = randomFrom(XContentType.values());
```

Comment:
Not, it's a leftover in the previous test, XContent type must be chosen randomly at the beginning of all tests. I changed that.